src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -48,8 +49,17 @@
     static {
         EXECUTORS = new ConcurrentHashMap<ScheduledThreadPoolExecutor, ScheduledThreadPoolExecutor>();
         PURGE = new AtomicReference<ScheduledExecutorService>();
-        PURGE_FORCE = Boolean.getBoolean(PURGE_FORCE_KEY);
         PURGE_FREQUENCY = Integer.getInteger(FREQUENCY_KEY, 1000);
+
+        // Forces the use of purge even if setRemoveOnCancelPolicy is available
+        final boolean purgeForce = Boolean.getBoolean(PURGE_FORCE_KEY); // Compiler should inline this
+
+        // According to http://developer.android.com/reference/java/util/concurrent/ScheduledThreadPoolExecutor.html#setRemoveOnCancelPolicy(boolean)
+        // setRemoveOnCancelPolicy available since Android API 21
+        final int androidApiVersion = PlatformDependent.getAndroidApiVersion();
+
+        SHOULD_TRY_ENABLE_CANCEL_POLICY = !purgeForce
+                        && (androidApiVersion == ANDROID_API_VERSION_IS_NOT_ANDROID || androidApiVersion >= 21);
```

Comment:
Four space indent is already killing me on this whole project, but continuations should be double that which is 8. This is 16.